gateleen-kafka/src/main/java/org/swisspush/gateleen/kafka/KafkaProducerRecordBuilder.java

```diff
@@ -52,33 +52,31 @@ class KafkaProducerRecordBuilder {
      * @throws ValidationException when the payload is not valid (missing properties, wrong types, etc.)
      */
     Future<List<KafkaProducerRecord<String, String>>> buildRecordsAsync(String topic, Buffer payload) {
-        return Future.<Void>succeededFuture().compose((Void v) -> {
+        return Future.<Void>succeededFuture().compose((Void v) -> vertx.executeBlocking(() -> {
+            long beginEpchMs = currentTimeMillis();
             JsonObject payloadObj;
             try {
                 payloadObj = new JsonObject(payload);
             } catch (DecodeException de) {
-                return Future.failedFuture(new ValidationException("Error while parsing payload", de));
+                throw new ValidationException("Error while parsing payload", de);
```

Comment:
Is the outcome really the same with returning a failed Future and throwing an Exception? I just found in vertx-rest-storage a place where we get an EventBus timeout when calling vertx-rest-storage from gateelen and there is an error in vertx-rest-storage. In this case, the EventBus message is not responded

Reply:
AFAIK its the way of how the newer "*executeBlocking*" works. But to be sure I will have a look how it actually behaves.

Reply:
I did the following experiment which seems to support the thesis that the code is correct.
Reproducer:
```java
    public static void main(String[] args) {
        try {
            Vertx vertx = Vertx.vertx();
            Future<Object> fut = vertx.executeBlocking(() -> {
                throw new RuntimeException("hello exceptional world");
            });
            fut.onFailure(ex -> {
                System.out.println("Whopsee: "+ ex.getMessage());
            });
            fut.onSuccess(obj -> {
                System.out.println("It was ok :)");
            });
        } catch (Exception ex) {
            System.out.println("The outer catch block");
        }
    }
```
Result:
```
Whopsee: hello exceptional world
```